modules/rntuple.mjs

```diff
@@ -747,35 +732,53 @@ class RNTupleDescriptorBuilder {
         this.pageLocations = clusterPageLocations;
     }
 
-    // Example Of Deserializing Page Content
-    deserializePage(blob, columnDescriptor) {
-        const originalColtype = columnDescriptor.coltype,
-        { coltype } = recontructUnsplitBuffer(blob, columnDescriptor);
-        let { blob: processedBlob } = recontructUnsplitBuffer(blob, columnDescriptor);
+// Example Of Deserializing Page Content
+deserializePage(blob, columnDescriptor, pageInfo) {
+    const originalColtype = columnDescriptor.coltype,
+    { coltype } = recontructUnsplitBuffer(blob, columnDescriptor);
+    let { blob: processedBlob } = recontructUnsplitBuffer(blob, columnDescriptor);
 
-        
-        // Handle split index types
+    
+    // Handle split index types
         if (originalColtype === ENTupleColumnType.kSplitIndex32 || originalColtype=== ENTupleColumnType.kSplitIndex64) {
-            const { blob: decodedArray } = DecodeDeltaIndex(processedBlob, coltype);
-            processedBlob = decodedArray;  
-        }
+        const { blob: decodedArray } = DecodeDeltaIndex(processedBlob, coltype);
+        processedBlob = decodedArray;  
+    }
 
-        // Handle Split Signed Int types
-        if (originalColtype === ENTupleColumnType.kSplitInt16 || originalColtype === ENTupleColumnType.kSplitInt32 || originalColtype === ENTupleColumnType.kSplitInt64) {
-            const { blob: decodedArray } = decodeZigzag(processedBlob, coltype);
-            processedBlob = decodedArray;  
-        }
+    // Handle Split Signed Int types
+    if (originalColtype === ENTupleColumnType.kSplitInt16 || originalColtype === ENTupleColumnType.kSplitInt32 || originalColtype === ENTupleColumnType.kSplitInt64) {
+        const { blob: decodedArray } = decodeZigzag(processedBlob, coltype);
+        processedBlob = decodedArray;  
+    }
 
-        const byteSize = getTypeByteSize(coltype),
-              reader = new RBufferReader(processedBlob),
-              values = [];
+    const reader = new RBufferReader(processedBlob),
+          values = [],
 
-        if (!byteSize)
-            throw new Error('Invalid or unsupported column type: cannot determine byte size');
+    // Use numElements from pageInfo parameter
+    numValues = Number(pageInfo.numElements);
 
-        const numValues = processedBlob.byteLength / byteSize;
+    switch (coltype) {
+        case ENTupleColumnType.kBit: {
+            let bitCount = 0;
+            const totalBitsInBuffer = processedBlob.byteLength * 8;
```

Comment:
You should not really need this value, as `totalBitsInBuffer` is always going to be `>= numValues` (so checking it in the `for` condition is redundant).
However you can explicitly validate this assumption by adding a check and throw if it's not the case